core/test/base/types.cpp

```diff
@@ -100,4 +105,109 @@ TEST(PrecisionReduction, ComputesCommonEncoding)
 }
 
 
+TEST(ConfigSet, MaskCorrectly)
+{
+    constexpr auto mask3_u = gko::detail::mask<3>();
+    constexpr auto fullmask_u = gko::detail::mask<32>();
+    constexpr auto mask3_u64 = gko::detail::mask<3, std::uint64_t>();
+    constexpr auto fullmask_u64 = gko::detail::mask<64, std::uint64_t>();
+
+    ASSERT_EQ(mask3_u, 7u);
+    ASSERT_EQ(fullmask_u, 0xffffffffu);
+    ASSERT_EQ((std::is_same<decltype(mask3_u), const unsigned int>::value),
+              true);
+    ASSERT_EQ((std::is_same<decltype(fullmask_u), const unsigned int>::value),
+              true);
```

Comment:
```suggestion
    ASSERT_TRUE((std::is_same<decltype(mask3_u), const unsigned int>::value));
    ASSERT_TRUE((std::is_same<decltype(fullmask_u), const unsigned int>::value));
```
Not sure if you even need the additional parentheses then?

Reply:
It still needs the additional parentheses such that the macro unpacks parameter correctly.